tests/test_tileserver.py

```diff
@@ -1,35 +1,79 @@
 """Tests for tileserver."""
 import pytest
+from click.testing import CliRunner
+from pathlib import Path
+import numpy as np
 
+from tiatoolbox import cli
 from tiatoolbox.visualization.tileserver import TileServer
 from tiatoolbox.wsicore.wsireader import WSIReader
 
 
 @pytest.fixture()
-def app(remote_sample) -> TileServer:
+def app(sample_ndpi) -> TileServer:
     """Create a testing TileServer WSGI app."""
-    wsi = WSIReader.open(remote_sample("ndpi-1"))
-    app = TileServer(title="test", layers={"ndpi": wsi})
+    runner = CliRunner()
+
+    # make a low-res .jpg of the right shape to be used as
+    # a low-res overlay
+    runner.invoke(
+        cli.main,
+        [
+            "slide-thumbnail",
+            "--img-input",
+            str(Path(sample_ndpi)),
+            "--mode",
+            "save",
+        ],
+    )
```

Comment:
Any particular reason why you are creating a Cli handler and then calling that rather than just using the slide_thumbnail method on WSIReader? Just seems like an odd way to make a thumbnail for the test.

Reply:
i think it was a historical artifact of something i tried and removed, cant quite remember now! but have changed it